tests_integration/test_git.py

```diff
@@ -0,0 +1,271 @@
+
+    try:
+        repository_path = f"@{git_repository}"
+        runner.invoke_with_connection(["git", "list-files", repository_path])
+    except ProgrammingError as err:
+        assert (
+            err.raw_msg
+            == "Files paths in git repositories must specify a scope. For example, a branch name, "
+            "a tag name, or a valid commit hash. Commit hashes are between 6 and 40 characters long."
+        )
+
+    try:
+        repository_path = f"@{git_repository}/branches/branch_which_does_not_exist"
+        runner.invoke_with_connection(["git", "list-files", repository_path])
+    except ProgrammingError as err:
+        assert (
+            err.raw_msg
+            == "The specified branch 'branch_which_does_not_exist' cannot be found in the Git Repository."
+        )
+
+    try:
+        repository_path = f"@{git_repository}/tags/tag_which_does_not_exist"
+        runner.invoke_with_connection(["git", "list-files", repository_path])
+    except ProgrammingError as err:
+        assert (
+            err.raw_msg
+            == "The specified tag 'tag_which_does_not_exist' cannot be found in the Git Repository."
+        )
+
+    # list-files - branch - no '/' at the end
+    repository_path = f"@{git_repository}/branches/{BRANCH_NAME}"
+    result = runner.invoke_with_connection_json(["git", "list-files", repository_path])
+    assert result.exit_code == 0
+    prefix = repository_path[1:].lower()
+    assert _filter_key(result.json, key="name") == [
+        f"{prefix}/an_existing_directory/file_inside.txt",
+        f"{prefix}/file.txt",
+        f"{prefix}/file_only_present_on_a_different_branch.txt",
+    ]
+
+    # list-files - tags - '/' at the end
+    repository_path = f"@{git_repository}/tags/{TAG_NAME}/"
+    result = runner.invoke_with_connection_json(["git", "list-files", repository_path])
+    assert result.exit_code == 0
+    prefix = repository_path[1:-1].lower()
+    assert _filter_key(result.json, key="name") == [
+        f"{prefix}/an_existing_directory/file_inside.txt",
+        f"{prefix}/file.txt",
+        f"{prefix}/file_only_present_on_a_particular_tag.txt",
+    ]
+
+
+@pytest.mark.integration
+def test_copy(runner, git_repository):
+    # create stage for testing copy
+    STAGE_NAME = "a_perfect_stage_for_testing"
+    result = runner.invoke_with_connection(["sql", "-q", f"create stage {STAGE_NAME}"])
+    assert result.exit_code == 0
+
+    # copy - test error messages
+    result = runner.invoke_with_connection(
+        ["git", "copy", git_repository, f"@{STAGE_NAME}"]
+    )
+    _assert_error_message_in_output(result.output)
+    try:
+        repository_path = f"@{git_repository}"
+        runner.invoke_with_connection(
+            ["git", "copy", repository_path, f"@{STAGE_NAME}"]
+        )
+    except ProgrammingError as err:
+        assert (
+            err.raw_msg
+            == "Files paths in git repositories must specify a scope. For example, a branch name, "
```

Comment:
These messages are from Snowflake, right? 
I'm not sure if we have to test their content. Do we gain any extra confidence thanks to this assertion? Or maybe just checking that some error occurred is enough? WDYT?

Reply:
We will have to adjust the assertions with every small change in these messages but I'm not sure how often that happens.

Reply:
I reduced number of errors, just to make sure thet they're visible toi the user